debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlConnectorIT.java

```diff
@@ -1983,4 +2012,54 @@ private void waitForStreamingRunning(String serverName) throws InterruptedExcept
         final List<SourceRecord> uc = records.recordsForTopic(RO_DATABASE.topicForTable("Products"));
         return uc != null ? uc : records.recordsForTopic(RO_DATABASE.topicForTable("products"));
     }
+
+    @Test
+    @FixFor("DBZ-1531")
+    public void shouldEmitHeadersOnPrimaryKeyUpdate() throws Exception {
+        config = DATABASE.defaultConfig()
+                .with(MySqlConnectorConfig.SNAPSHOT_MODE, MySqlConnectorConfig.SnapshotMode.NEVER)
+                .build();
+
+        // Start the connector ...
+        start(MySqlConnector.class, config);
+
+        // ---------------------------------------------------------------------------------------------------------------
+        // Consume all of the events due to startup and initialization of the database
+        // ---------------------------------------------------------------------------------------------------------------
+        SourceRecords records = consumeRecordsByTopic(INITIAL_EVENT_COUNT); // 6 DDL changes
+        assertThat(records.recordsForTopic(DATABASE.topicForTable("orders")).size()).isEqualTo(5);
+
+        try (MySQLConnection db = MySQLConnection.forTestDatabase(DATABASE.getDatabaseName());) {
+            try (JdbcConnection connection = db.connect()) {
+                connection.execute("UPDATE orders SET order_number=10303 WHERE order_number=10003");
+            }
+        }
+        // Consume the update of the PK, which is one insert followed by a delete followed by a tombstone ...
+        records = consumeRecordsByTopic(3);
+        List<SourceRecord> updates = records.recordsForTopic(DATABASE.topicForTable("orders"));
+
+        assertThat(updates.size()).isEqualTo(3);
+
+        SourceRecord deleteRecord = updates.get(0);
+        Header keyPKUpdateHeader = getPKUpdateNewKeyHeader(deleteRecord);
+        assertEquals(Integer.valueOf(10303), ((Struct) keyPKUpdateHeader.value()).getInt32("order_number"));
+
+        SourceRecord insertRecord = updates.get(2);
+        keyPKUpdateHeader = getPKUpdateOldKeyHeader(insertRecord);
+        assertEquals(Integer.valueOf(10003), ((Struct) keyPKUpdateHeader.value()).getInt32("order_number"));
+
+        try (MySQLConnection db = MySQLConnection.forTestDatabase(DATABASE.getDatabaseName());) {
+            try (JdbcConnection connection = db.connect()) {
+                connection.execute("UPDATE orders SET quantity=5 WHERE order_number=10004");
+            }
+        }
+        records = consumeRecordsByTopic(5);
+        updates = records.recordsForTopic(DATABASE.topicForTable("orders"));
+        assertThat(updates.size()).isEqualTo(1);
+
+        SourceRecord updateRecord = updates.get(0);
```

Comment:
That's unused. I'll expand a bit to assert the header isn't present for this record.